src/power_grid_model/utils.py

```diff
@@ -301,3 +304,61 @@ def _compatibility_deprecated_import_json_data(json_file: Path, data_type: Datas
         raise PowerGridSerializationError("An internal error occured during deserialization")
 
     return result
+
+
+def self_test():
+    """
+    Perform a self-test of the Power Grid Model functionality to check if the installation was successful
+    and there are no build errors, segfaults, undefined symbols, etc.
+
+    This function is designed to validate the basic functionality of data serialization,
+    model instantiation, power flow calculation, and serialization of results using the
+    Power Grid Model library.
+
+    Raises:
+        PowerGridSerializationError: if an internal error occured during deserialization.
+        AssertionError: If the output data is empty or if the calculation results are incorrect.
+        PowerGridError: if there was an internal error.
```

Comment:
I believe these two errors are not raised anywhere (yet?) within `self_test()`.

Reply:
`assert` raises an `AssertionError`. `PowerGridError` may be used anywhere in the calls to the PGM core and indeed are not explicitly raised in this function. However, since this entire function is intended to test the PGM core in a single command, anything may go wrong and the user will not know what, except if we passthrough the exception.

If you prefer, we can make a `try`/`except` block around the PGM calls like so:

```py
def self_test():
    # setup stuff
    try:
        input_data = deserialize_from_file(...)
        model = PowerGridModel(input_data)
        # etc
    except PowerGridError:
        raise  # explicitly re-throw
    # winddown
```

Reply:
I see. I didn't realize that `assert` is raising the error itself.

On the other hand, I think that explicitly re-throwing the errors as you suggested is probably the way to go. The users will know exactly what happened, as if there is a bug, we would also be able to debug it more easily if we have access to a more detailed log.